state/prune.go

```diff
@@ -117,6 +155,33 @@ func (p *collectionPruner) pruneByAge() error {
 	return nil
 }
 
+func (*collectionPruner) toDeleteCalculator(coll *mgo.Collection, maxSize int, countRatio float64) (int, error) {
+	collMB, err := getCollectionMB(coll)
+	if err != nil {
+		return 0, errors.Annotate(err, "retrieving collection size")
+	}
+	if collMB <= maxSize {
+		return 0, nil
+	}
+	count, err := coll.Count()
+	if err == mgo.ErrNotFound || count <= 0 {
+		return 0, nil
+	}
+	if err != nil {
+		return 0, errors.Annotatef(err, "counting %s records", coll.Name)
+	}
+	// We are making the assumption that item sizes can be averaged for
```

Comment:
Consider revising:
// For large numbers of items we are making an assumption that the size of items can be averaged to give a reasonable number of items to drop to reach the goal size.

Reply:
ok